esphome/components/climate_mitsubishi/climate_mitsubishi.cpp

```diff
@@ -0,0 +1,795 @@
+                                     climate::CLIMATE_MODE_HEAT_COOL});
+  this->traits_.set_supported_fan_modes({climate::CLIMATE_FAN_AUTO, climate::CLIMATE_FAN_QUIET});
+  this->traits_.set_supported_custom_fan_modes({"1", "2", "3", "4"});
+  this->traits_.set_supports_current_temperature(true);
+  this->traits_.set_supports_action(true);
+  this->traits_.set_visual_current_temperature_step(0.1);
+  this->traits_.set_visual_target_temperature_step(0.5);
+  this->pending_requests_ = std::list<RequestSlot *>();
+}
+
+ClimateTraits ClimateMitsubishi::traits() { return traits_; }
+
+ClimateMode ClimateMitsubishi::mode_to_climate_mode_(uint8_t mode) {
+  switch (mode) {
+    case (uint8_t) mitsubishi_protocol::Mode::HEAT:
+      return ClimateMode::CLIMATE_MODE_HEAT;
+      break;
+    case (uint8_t) mitsubishi_protocol::Mode::COOL:
+      return ClimateMode::CLIMATE_MODE_COOL;
+      break;
+    case (uint8_t) mitsubishi_protocol::Mode::DRY:
+      return ClimateMode::CLIMATE_MODE_DRY;
+      break;
+    case (uint8_t) mitsubishi_protocol::Mode::FAN:
+      return ClimateMode::CLIMATE_MODE_FAN_ONLY;
+      break;
+    case (uint8_t) mitsubishi_protocol::Mode::AUTO:
+      return ClimateMode::CLIMATE_MODE_HEAT_COOL;
+      break;
+    default:
+      return ClimateMode::CLIMATE_MODE_OFF;
+      break;
+  }
+}
+
+std::string ClimateMitsubishi::fan_to_custom_fan_mode_(uint8_t fan) {
+  switch (fan) {
+    case (uint8_t) mitsubishi_protocol::FanMode::AUTO:
+      return "Auto";
+      break;
+    case (uint8_t) mitsubishi_protocol::FanMode::FAN_QUIET:
+      return "Quiet";
+      break;
+    case (uint8_t) mitsubishi_protocol::FanMode::FAN_1:
+      return "1";
+      break;
+    case (uint8_t) mitsubishi_protocol::FanMode::FAN_2:
+      return "2";
+      break;
+    case (uint8_t) mitsubishi_protocol::FanMode::FAN_3:
+      return "3";
+      break;
+    case (uint8_t) mitsubishi_protocol::FanMode::FAN_4:
+      return "4";
+      break;
+    default:
+      return "Unknown";
+      break;
+  }
+}
+
+std::string ClimateMitsubishi::vertical_vane_to_vertical_airflow_select_(uint8_t vertical_vane) {
+  switch (vertical_vane) {
+    case (uint8_t) mitsubishi_protocol::VerticalVaneMode::VANE_AUTO:
+      return "Auto";
+    case (uint8_t) mitsubishi_protocol::VerticalVaneMode::VANE_1:
+      return "1";
+    case (uint8_t) mitsubishi_protocol::VerticalVaneMode::VANE_2:
+      return "2";
+    case (uint8_t) mitsubishi_protocol::VerticalVaneMode::VANE_3:
+      return "3";
+    case (uint8_t) mitsubishi_protocol::VerticalVaneMode::VANE_4:
+      return "4";
+    case (uint8_t) mitsubishi_protocol::VerticalVaneMode::VANE_5:
+      return "5";
+    case (uint8_t) mitsubishi_protocol::VerticalVaneMode::VANE_SWING:
+      return "Swing";
+    default:
+      return "Auto";
+      break;
+  }
+}
+
+std::string ClimateMitsubishi::horizontal_vane_to_horizontal_airflow_select_(uint8_t vertical_vane) {
+  switch (vertical_vane) {
```

Comment:
Fix for proper reading Wide Vane setting from the unit
```suggestion
  switch (vertical_vane & 0x0F) {
```

There is apparently some units which expect values prepended with 0x80: https://github.com/SwiCago/HeatPump/commit/f5755b1daa36d105278b8a1b6e0fb01225b48cad